smarts/sstudio/types.py

```diff
@@ -565,21 +542,27 @@ def pick_remaining_shape_after_split(geometry_collection, length, lane):
 
             min_cut = min(lane_offset, lane_length)
             # Second cut takes into account shortening of geometry by `min_cut`.
-            max_cut = min(min_cut + geom_length, lane_length - min_cut)
+            max_cut = min(min_cut + geom_length, lane_length)
 
             lane_shape = road_network.split_lane_shape_at_offset(
                 lane_shape, lane, min_cut
             )
-            lane_shape = pick_remaining_shape_after_split(lane_shape, min_cut, lane)
-            if lane_shape is None:
-                continue
+            if isinstance(lane_shape, GeometryCollection):
+                if len(lane_shape) not in {1, 2}:
+                    continue
+                else:
+                    lane_shape = lane_shape[0]
```

Comment:
lane_shape[0] contains the right half of the splitted shape, lane_shape[1] contains the left half of the splitted.
Here is selecting the right half splitted by min_cut.